spec.bs

```diff
@@ -2932,6 +2940,193 @@ This specification defines two [=policy-controlled features=] identified by the
+: <dfn>ad slot</dfn>
+:: A [=string=]. Matches the `adSlot` key of the JSON dictionaries in the top-level array of the
+  [:Ad-Auction-Signals:] value.
+
+</dl>
+
+<h3 id=direct-from-sellers-signals-struct>Direct from seller signals</h3>
+A <dfn>direct from seller signals</dfn> is a [=struct=] with the following [=struct/items=]:
+
+NOTE: This is only captured during a [=request=] whose [=request/initiator type=] is `"fetch"`, made
+with the {{RequestInit/adAuctionHeaders}} option set to `true`, as described in the
+[:Ad-Auction-Signals:] header description.
+
+<dl dfn-for="direct from seller signals">
+: <dfn>auction signals</dfn>
+:: Null or a [=string=].
+  Opaque JSON data passed to both buyers' and the seller's [=script runners=].
+: <dfn>seller signals</dfn>
+:: Null or a [=string=].
+  Opaque JSON data passed to the seller's [=script runner=].
+: <dfn>per buyer signals</dfn>
+:: A [=map=] whose [=map/keys=] are [=origins=] and whose [=map/values=] are [=strings=].
+  [=map/Keys=] are buyers and must be valid HTTPS origins. [=map/Values=] are opaque JSON data
+  passed to corresponding buyer's [=script runner=].
+
+</dl>
+
+Each [=traversable navigable=] has a <dfn for="traversable navigable">captured ad auction headers
+</dfn>, which is a [=map=] whose [=map/keys=] are [=direct from seller signals keys=] and whose
+[=map/values=] are [=direct from seller signals=].
+</div>
+
+<div algorithm="fetch auction headers patch">
+The following will be added to [[Fetch#http-extensions]]:
+
+<h3 id=sec-ad-auction-fetch-header>The \`<a http-header><code>Sec-Ad-Auction-Fetch</code></a>\` HTTP
+request header</h3>
+
+The \`<dfn http-header><code>Sec-Ad-Auction-Fetch</code></dfn>\` request header is an optional
+[=structured header=] with of type [=structured header/boolean=]. [:Sec-Ad-Auction-Fetch:] will only
+be set on a [=request=] whose [=request/initiator type=] is `"fetch"`, made with the
+{{RequestInit/adAuctionHeaders}} option set to `true`. If [:Sec-Ad-Auction-Fetch:] is equal to `?1`,
+the user agent will remove any [:Ad-Auction-Signals:] from the returned [=response=] -- the
+[:Ad-Auction-Signals:] value will instead only be used in Protected Audiences auctions. 
+
+<h3 id=ad-auction-signals-header>The \`<a http-header><code>Ad-Auction-Signals</code></a>\` HTTP
+response header</h3>
+
+The \`<dfn http-header><code>Ad-Auction-Signals</code></dfn>\` response header provides value
+of a JSON array of dictionaries, each with an `adSlot` key. Protected Audience non-component,
+component, and top-level auctions may specify which signals to load by the `adSlot` key.
+</div>
+
+<div algorithm="fetch auction signals response patch">
+The following step will be added to the [=HTTP-network-or-cache fetch=] algorithm, before step
+"Return |response|.":
+
+1. If |response| is not null, |fetchParams|'s [=fetch params/task destination=] is a [=global
+  object=], and |request|'s [=request/capture-ad-auction-headers=] is `true`, then run [=update

```

Comment:
Sorry for just catching this now, but this should be `is a [=global object=] that's a {{Window}} object`. See for example: https://fetch.spec.whatwg.org/#:~:text=If%20request%E2%80%99s%20window%20is%20%22client%22%2C%20then%20set%20request%E2%80%99s%20window%20to%20request%E2%80%99s%20client%2C%20if%20request%E2%80%99s%20client%E2%80%99s%20global%20object%20is%20a%20Window%20object%3B%20otherwise%20%22no%2Dwindow. As it stands right now, service workers that try and make these special fetches will try and find their "document"'s traversable navigable, which is impossible since their global isn't a Window, and therefore doesn't have an `[=associated Document=]`.

Reply:
Done.